proof_ipa.go

```diff
@@ -332,7 +332,8 @@ func DeserializeProof(vp *VerkleProof, statediff StateDiff) (*Proof, error) {
 
 	poaStems = make([][]byte, len(vp.OtherStems))
 	for i, poaStem := range vp.OtherStems {
-		poaStems[i] = poaStem[:]
```

Comment:
Doing `poaStem[:]` was incorrect. It's taking the `poaStem` **array** as a slice. But `poaStem` is always the same variable in the loop, so all `poaStems` items will be pointing to the same value.

The effect this had is that all `poaStems` returned have the same value.